app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -607,17 +634,27 @@ private void updateDrawerNavigation() {
         final DrawerLayout drawer = findViewById(R.id.drawer_layout);
 
         final Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.fragment_holder);
+
+        //If it's MainFragment enable the Navigation Drawer again
         if (fragment instanceof MainFragment) {
             getSupportActionBar().setDisplayHomeAsUpEnabled(false);
             if (toggle != null) {
                 toggle.syncState();
                 toolbar.setNavigationOnClickListener(v -> drawer.openDrawer(GravityCompat.START));
                 drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNDEFINED);
             }
-        } else {
-            drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
-            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-            toolbar.setNavigationOnClickListener(v -> onHomeButtonPressed());
+            return;
+        }
+
+        //We need to enable the home button for the search fragment to go back to MainFragment
+        if (fragment instanceof SearchFragment) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(false);
+            if (toggle != null) {
+                drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
+                getSupportActionBar().setDisplayHomeAsUpEnabled(true);
```

Comment:
Question: why are you first setting `HomeAsUpEnabled` to `false` and then to `true`?

Reply:
Changed it :) my bad